pkg/apis/core/validation/validation.go

```diff
@@ -2957,6 +2957,18 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
 		for _, msg := range ValidateServiceAccountName(spec.ServiceAccountName, false) {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceAccountName"), spec.ServiceAccountName, msg))
 		}
+	} else {
+		for vi, volume := range spec.Volumes {
+			path := fldPath.Child("volumes").Index(vi).Child("projected")
+			if volume.Projected != nil {
+				for si, source := range volume.Projected.Sources {
+					saPath := path.Child("sources").Index(si).Child("serviceAccountToken")
+					if source.ServiceAccountToken != nil {
+						allErrs = append(allErrs, field.Invalid(saPath, source.ServiceAccountToken, "serviceaccount name is empty, may not specify serviceAccountToken projected volume"))
```

Comment:
* use field.Forbidden instead of field.Invalid
* error message: "must not be specified when serviceAccountName is not set"